core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -487,4 +339,163 @@ public boolean isFinished()
             return current == null && finished;
         }
     }
+
+    private class ChannelFilter
+    {
+        private final int channelIndex;
+
+        private ChannelState state;
+        private boolean minMaxPossible;
+        @Nullable
+        private BlockBuilder blockBuilder;
+        @Nullable
+        private TypedSet valueSet;
+        @Nullable
+        private Block minValues;
+        @Nullable
+        private Block maxValues;
+        @Nullable
+        private BlockPositionComparison minMaxComparison;
+
+        public ChannelFilter(Channel channel, boolean minMaxEnabled, PlanNodeId planNodeId, BlockTypeOperators blockTypeOperators, int channelIndex)
```

Comment:
We don't need to make any method in this class `public`

Reply:
I prefer to make methods that are used outside of this class public, for readability reasons.